src/component/Tooltip.tsx

```diff
@@ -90,82 +90,78 @@ export type TooltipProps<TValue extends ValueType, TName extends NameType> = Def
   useTranslate3d?: boolean;
 };
 
-export class Tooltip<TValue extends ValueType, TName extends NameType> extends PureComponent<
-  TooltipProps<TValue, TName>
-> {
-  static displayName = 'Tooltip';
-
-  static defaultProps = {
-    active: false,
-    allowEscapeViewBox: { x: false, y: false },
-    reverseDirection: { x: false, y: false },
-    offset: 10,
-    viewBox: { x1: 0, x2: 0, y1: 0, y2: 0 },
-    coordinate: { x: 0, y: 0 },
-    cursorStyle: {},
-    separator: ' : ',
-    wrapperStyle: {},
-    contentStyle: {},
-    itemStyle: {},
-    labelStyle: {},
-    cursor: true,
-    trigger: 'hover',
-    isAnimationActive: !Global.isSsr,
-    animationEasing: 'ease',
-    animationDuration: 400,
-    filterNull: true,
-    useTranslate3d: false,
-  };
+const tooltipDefaultProps = {
+  active: false,
+  allowEscapeViewBox: { x: false, y: false },
+  reverseDirection: { x: false, y: false },
+  offset: 10,
+  viewBox: { x: 100, y: 100, height: 50, width: 50 },
+  coordinate: { x: 0, y: 0 },
+  wrapperStyle: {},
+  // this doesn't exist on TooltipProps
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  cursorStyle: {},
+  separator: ' : ',
+  contentStyle: {},
+  itemStyle: {},
+  labelStyle: {},
+  isAnimationActive: !Global.isSsr,
+  animationEasing: 'ease',
+  animationDuration: 400,
+  filterNull: true,
+  useTranslate3d: false,
+  cursor: true,
+};
 
-  state = {
+const Tooltip = <TValue extends ValueType, TName extends NameType>(
+  props: TooltipProps<TValue, TName> & { children?: React.ReactNode },
+) => {
+  const [state, setState] = useState({
```

Comment:
While this works, it is recommended to split the state into separate variables. 
https://reactjs.org/docs/hooks-faq.html#should-i-use-one-or-many-state-variables

Reply:
agreed, was doing this for the sake of speed when originally refactoring. Can revise